docs/modules/ROOT/pages/api/account.adoc

```diff
@@ -0,0 +1,213 @@
+--
+
+[.contract-index]
+.Events
+--
+* xref:#Account-OwnerAdded[`++OwnerAdded(new_owner_guid)++`]
+* xref:#Account-OwnerRemoved[`++OwnerRemoved(removed_owner_guid)++`]
+--
+
+[.contract-item]
+[[Account-__validate_deploy__]]
+==== `[.contract-item-name]#++__validate_deploy__++#++(self: @ContractState, class_hash: felt252, contract_address_salt: felt252, _public_key: felt252) → felt252++` [.item-kind]#external#
+
+Validates a DeployAccount transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-__execute__]]
+==== `[.contract-item-name]#++__execute__++#++(ref self: ContractState, calls: Array<Call>) → Array<Span<felt252>>++` [.item-kind]#external#
+
+See xref:ISRC6-\\__execute__[ISRC6::\\__execute__].
+
+[.contract-item]
+[[Account-__validate__]]
+==== `[.contract-item-name]#++__validate__++#++(self: @ContractState, calls: Array<Call>) → felt252++` [.item-kind]#external#
+
+See xref:ISRC6-\\__validate__[ISRC6::\\__validate__].
+
+[.contract-item]
+[[Account-is_valid_signature]]
+==== `[.contract-item-name]#++is_valid_signature++#++(self: @ContractState, hash: felt252, signature: Array<felt252>) → felt252++` [.item-kind]#external#
+
+See xref:ISRC6-is_valid_signature[ISRC6::is_valid_signature].
+
+[.contract-item]
+[[Account-supports_interface]]
+==== `[.contract-item-name]#++supports_interface++#++(self: @ContractState, interface_id: felt252) → bool++` [.item-kind]#external#
+
+Returns whether a contract implements a given interface or not.
+
+[.contract-item]
+[[Account-__validate_declare__]]
+==== `[.contract-item-name]#++__validate_declare__++#++(self: @ContractState, class_hash: felt252) → felt252++` [.item-kind]#external#
+
+Validates a Declare transaction.
+
+Returns the short string `'VALID'` if valid, or any other value otherwise.
+
+[.contract-item]
+[[Account-set_public_key]]
+==== `[.contract-item-name]#++set_public_key++#++(ref self: ContractState, new_public_key: felt252)++` [.item-kind]#external#
+
+Sets a new public key for the account. Can be called only by self through the `\\__execute__` mechanism.
+
+Emits an {OwnerRemoved}, and an {OwnerAdded} event.
+
+[.contract-item]
+[[Account-get_public_key]]
+==== `[.contract-item-name]#++get_public_key++#++(self: @ContractState)++ → felt252` [.item-kind]#external#
+
+Returns the current public key of the account.
+
+[.contract-item]
+[[Account-initializer]]
+==== `[.contract-item-name]#++initializer++#++(ref self: ContractState, _public_key: felt252)++` [.item-kind]#internal#
+
+Initializes the account with the given public key, and registers the ISRC6 interface ID.
+
+Emits an {OwnerAdded} event.
+
+[.contract-item]
+[[Account-validate_transaction]]
+==== `[.contract-item-name]#++validate_transaction++#++(self: @ContractState)++ → felt252` [.item-kind]#internal#
+
+Validates a transaction signature from the global context.
+
+Returns the short string `'VALID'` if valid.
```

Comment:
```suggestion
Returns the short string `'VALID'` if valid, otherwise it reverts.
```
it's important to state when a function reverts, since it's a very relevant part of the interface